pkg/services/ngalert/notifier/multiorg_alertmanager.go

```diff
@@ -78,49 +82,119 @@ func (moa *MultiOrgAlertmanager) LoadAndSyncAlertmanagersForOrgs(ctx context.Con
 	return nil
 }
 
+//getLatestConfigs retrieves the latest AlertManager configuration for every organization and returns them as a map where the key is ID of an organization
+func (moa *MultiOrgAlertmanager) getLatestConfigs() (map[int64]*models.AlertConfiguration, error) {
+	configs, err := moa.configStore.GetAllLatestAlertmanagerConfiguration()
+	if err != nil {
+		return nil, err
+	}
+
+	result := make(map[int64]*models.AlertConfiguration, len(configs))
+	for _, config := range configs {
+		result[config.OrgID] = config
+	}
+
+	return result, nil
+}
+
 func (moa *MultiOrgAlertmanager) SyncAlertmanagersForOrgs(orgIDs []int64) {
 	orgsFound := make(map[int64]struct{}, len(orgIDs))
 	moa.alertmanagersMtx.Lock()
+
+	moa.addOrUpdateAlertManagerConfigurations(orgIDs, orgsFound, func(orgID int64) (*Alertmanager, error) {
+		// These metrics are not exported by Grafana and are mostly a placeholder.
+		// To export them, we need to translate the metrics from each individual registry and,
+		// then aggregate them on the main registry.
+		m := metrics.NewAlertmanagerMetrics(moa.metrics.GetOrCreateOrgRegistry(orgID))
+		return newAlertmanager(orgID, moa.settings, moa.configStore, moa.kvStore, m)
+	})
+
+	amsToStop := moa.cleanUpAlertManagers(orgsFound)
+
+	moa.metrics.ActiveConfigurations.Set(float64(len(moa.alertmanagers)))
+	moa.alertmanagersMtx.Unlock()
+
+	// Now, we can stop the Alertmanagers without having to hold a lock.
+	for orgID, am := range amsToStop {
+		moa.logger.Info("stopping Alertmanager", "org", orgID)
+		am.StopAndWait()
+		moa.logger.Info("stopped Alertmanager", "org", orgID)
+	}
+}
+
+func (moa *MultiOrgAlertmanager) addOrUpdateAlertManagerConfigurations(orgIDs []int64, foundOrgs map[int64]struct{}, managerFactory alertManagerFactory) {
```

Comment:
I think this function should not be called without the mutex right? If so can we add a doc comment for this?